experimental/dds/tree2/src/shared-tree/sharedTreeChangeCodecs.ts

```diff
@@ -0,0 +1,81 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { IJsonCodec, ICodecOptions, ICodecFamily, makeCodecFamily } from "../codec";
+import { Mutable } from "../util";
+import {
+	ModularChangeset,
+	makeModularChangeCodec,
+	makeSchemaChangeCodec,
+} from "../feature-libraries";
+import { SharedTreeChange } from "./sharedTreeChangeTypes";
+
+// These can't be an interfaces or they don't get the special string indexer bonus property.
+// eslint-disable-next-line @typescript-eslint/consistent-type-definitions
+type EncodedModularChange = {
+	type: "data";
+	change: ReturnType<ReturnType<typeof makeModularChangeCodec>["encode"]>;
+};
+
+// eslint-disable-next-line @typescript-eslint/consistent-type-definitions
+type EncodedSchemaChange = {
+	type: "schema";
+	change: ReturnType<ReturnType<typeof makeModularChangeCodec>["encode"]>;
+};
+
+export interface EncodedSharedTreeChange {
+	readonly encodedChanges: readonly (EncodedModularChange | EncodedSchemaChange)[];
```

Comment:
Give the drive to keep our encoded format small, I have to ask: do we really benefit from naming this `encodedChanges` instead of `changes`?